src/unix/fs.c

```diff
@@ -389,12 +390,31 @@ static ssize_t uv__fs_pathmax_size(const char* path) {
 
   return pathmax;
 }
+#endif
 
 static ssize_t uv__fs_readlink(uv_fs_t* req) {
   ssize_t len;
   char* buf;
+  struct stat st;
+  int ret;
+
+  ret = lstat(req->path, &st);
+  if (ret != 0) {
+    return -1;
+  }
+  len = st.st_size;
```

Comment:
This seems like it could potentially lead to a very large allocation if the target file is not actually a symlink?